src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -646,44 +648,311 @@ private void JSON_response(HttpServletResponse response, String jsonValue,
 			log.error("Missing box integration parameters");
 			return null;
 		}
-		String remoteUserEmail = request.getRemoteUser();
-		
-		if (BoxUtils.getBoxAccessToken() == null)
-		{
+		String remoteUserEmail = request.getRemoteUser() + "@umich.edu";
```

Comment:
If friend account user logs in, the "request.getRemoteUser()" return value would be his email address registered with friend account. So the solution is to detect whether "request.getRemoteUser()" return value is an email address or not. If not, attach "@umich.edu" to it.